lib/big_query.ex

```diff
@@ -6,8 +6,8 @@ defmodule BigQuery do
   alias BigQuery.Impressions, as: Impressions
 
   defmodule Interval do
-    @enforce_keys [:from, :to, :seconds]
-    defstruct [:from, :to, :seconds]
+    @enforce_keys [:from, :to, :rollup]
+    defstruct [:from, :to, :rollup]
```

Comment:
Instead of this struct having `seconds: 3600`, it now has `rollup: BigQuery.TimestampRollups.Hourly`.

It seemed cleaner to encapsulate all the rollup-interval logic into these submodules.  They know how to calculate floors/ceilings of dates, compute ranges, return the SQL needed to execute, etc.

Reply:
👍